PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/CreateFunctionAction.java

```diff
@@ -0,0 +1,214 @@
+
+            PsiDirectory directory = com.intellij.psi.util.ClassUtil.sourceRoot(psiDirectory);
+            String newName = packageName.replace('.', '/');
+            final FunctionTemplate bindingTemplate;
+            try {
+                bindingTemplate = form.getFunctionTemplate(triggerType);
+                final String functionClassContent = AzureFunctionsUtils.substituteParametersInTemplate(bindingTemplate, parameters);
+                Application application = ApplicationManager.getApplication();
+                application.runWriteAction(() -> {
+                    CreateFileAction.MkDirs mkDirs = ApplicationManager.getApplication().runWriteAction((Computable<CreateFileAction.MkDirs>) () -> new CreateFileAction.MkDirs(newName + '/' + className, directory));
+                    PsiFileFactory factory = PsiFileFactory.getInstance(project);
+                    CommandProcessor.getInstance().executeCommand(project, () -> {
+                        PsiFile psiFile = factory.createFileFromText(className
+                                + ".java", JavaFileType.INSTANCE, functionClassContent);
+
+                        psiElements.add(mkDirs.directory.add(psiFile));
+                    }, null, null);
+
+                    if (StringUtils.equalsIgnoreCase(triggerType, CreateFunctionForm.EVENT_HUB_TRIGGER)) {
+                        String connectionString = getEventHubNamespaceConnectionString(form.getEventHubNamespace());
+                        saveLocalSetting(project, parameters.get("connection"), connectionString);
+                    }
+                });
+
+
+            } catch (AzureExecutionException e) {
+                PluginUtil.displayErrorDialogAndLog("Create Azure Function Class error", e.getMessage(), e);
+            }
+        }
+        if (!psiElements.isEmpty()) {
+            FileEditorManager.getInstance(project).openFile(psiElements.get(0).getContainingFile().getVirtualFile(), false);
+        }
+        return psiElements.toArray(new PsiElement[0]);
+    }
+    private String getEventHubNamespaceConnectionString(EventHubNamespace eventHubNamespace) {
+        try {
+            Azure azure = AuthMethodManager.getInstance().getAzureClient(eventHubNamespace.id().split("/")[2]);
+            EventHubNamespaceAuthorizationRule eventHubNamespaceAuthorizationRule = azure.eventHubNamespaces().
+                    authorizationRules().getByName(eventHubNamespace.resourceGroupName(), eventHubNamespace.name(),
+                    "RootManageSharedAccessKey");
+            return eventHubNamespaceAuthorizationRule.getKeys().primaryConnectionString();
+        } catch (IOException e) {
+            return null;
+        }
+    }
+    private static final String DEFAULT_LOCAL_SETTINGS_JSON = "{\"IsEncrypted\":false,\"Values\":{\"AzureWebJobsStorage\":\"\",\"FUNCTIONS_WORKER_RUNTIME\":\"java\"}}";
+
+    public void saveLocalSetting(Project project, String key, String value) {
```

Comment:
We could add some utils method in `FunctionUtils` like `saveAppSettingsToLocalSettings` for it was invoked in multi-place, like AppSettingsTable and Function local run